src/positions/saga.ts

```diff
@@ -214,23 +220,41 @@ export function* triggerShortcutSaga({ payload }: ReturnType<typeof triggerShort
         headers: {
           'Content-Type': 'application/json',
         },
-        body: JSON.stringify(payload),
+        body: JSON.stringify(payload.data),
       }
     )
     if (!response.ok) {
       throw new Error(`Unable to trigger shortcut: ${response.status} ${response.statusText}`)
     }
 
     const { data } = yield call([response, 'json'])
+    yield put(triggerShortcutSuccess({ id: payload.id, transactions: data.transactions }))
+  } catch (error) {
+    yield put(triggerShortcutFailure(payload.id))
+  }
+}
+
+export function* executeShortcutSaga({ payload }: ReturnType<typeof executeShortcut>) {
+  Logger.debug(`${TAG}/executeShortcutSaga`, 'Initiating execute shortcut')
 
+  try {
     const kit: ContractKit = yield call(getContractKit)
     const walletAddress: string = yield call(getConnectedUnlockedAccount)
     const normalizer = new TxParamsNormalizer(kit.connection)
 
-    Logger.debug(`${TAG}/triggerShortcutSaga`, 'Starting to claim reward(s)', data.transactions)
+    const triggeredShortcuts: TriggeredShortcuts = yield select(triggeredShortcutsStatusSelector)
+    // use JSON stringify / parse, otherwise the transaction fails with this
+    // error: 'Gas estimation failed: Could not decode transaction failure
+    // reason or Error: invalid argument 0: json: cannot unmarshal non-string
+    // into Go struct field TransactionArgs.chainId of type *hexutil.Big'
+    const shortcutTransactions = JSON.parse(
+      JSON.stringify(triggeredShortcuts[payload].transactions)
+    )
```

Comment:
Weird, now I wanna now what's happening :D

Reply:
yes same!!! but in the interest of time i parked it for now :/ it's reproducible for me every time...